teacher.rb

```diff
@@ -0,0 +1,12 @@
+class Teacher < Person
+  def initialize(specialization = 'Physics')
+    super()
+    @specialization = specialization
+  end
+
+  attr_reader :specialization
+
```

Comment:
- [ ] The teacher constructor extends parent's constructor by adding `@specialization` and a parameter for it, something like so: 
```ruby
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```
Checkout this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) and kindly fix that.

Reply:
Thanks for your guidance. I really appreciate it. I am going to make these changes.